### ISSUE TITLE
How to apply padding mask without using casual mask?

### ISSUE THREAD

**Issue by GXY0116** [Dao-AILab/flash-attention#1063]
Hi, thanks for the great work. I want to know if I don't want to use the casual mask, how to do the padding mask in FA2?
If I don't need the casual mask, but the data are padded in a batch, which function in FA2 should I use?

**Comment by tridao** [comment]
https://github.com/Dao-AILab/flash-attention/blob/74b0761ff7efc7b90d4e5aeb529c1b2a09a7458c/flash_attn/flash_attn_interface.py#L1053

**Comment by tridao** [comment]
no padding

**Comment by GXY0116** [comment]
> no padding

Thanks. For example, if the tokenized data is like torch.tensor([[1,1,1,1,0,0,0],[1,1,1,1,1,0,0]]), 0 is the padding token, but I don't want to use casual mask. If I use the 'flash_attn_varlen_func' function, which parameter in this function indicates the padding infomation of the tensor?
